load/scan.go

```diff
@@ -93,82 +95,108 @@ func scanWithIndexer(channels []*duplexChannel, batchSize uint, limit uint64, br
 		panic("--batch-size cannot be less than 1")
 	}
 
-	// Current batches (per channel) that are being filled
-	batches := make([]Batch, numChannels)
-	for i := range batches {
-		batches[i] = factory.New()
+	// Batches details
+	// 1. fillingBatches contains batches that are being filled with items from scanner.
+	//    As soon a batch has batchSize items in it, or there is no more items to come, batch moves to
+	// 2. unsentBatches, which contains batches, ready to be sent to worker.
+	//    As soon as there is place in chan to worker, batch is placed into chan
```

Comment:
This is a little unclear. Maybe something like:

```
2. unsentBatches contains batches ready to be sent to a worker. 
As soon as a worker's chan is available (i.e., not blocking), the batch is placed onto that worker's chan.
```

Reply:
Yes, agree, this part of text is somewhat clumsy. Need to be clarified.
PR updated.